src/compiler/sys.ts

```diff
@@ -30,36 +30,45 @@ namespace ts {
         mtime?: Date;
     }
 
-    export interface System {
-        args: string[];
+    /**
+     * Partial interface of the System thats needed to support the caching of directory structure
+     */
+    export interface PartialSystem {

```

Comment:
we need a better name for this one.

Reply:
`DirectoryStructureCacheHost`?

Reply:
Renamed this as DirectoryStructureHost and CachedPartialSystem as CachedDirectoryStructureHost